code/modules/mob/living/silicon/ai/ai_say.dm

```diff
@@ -49,108 +49,103 @@
-	log_talk(message, LOG_SAY, tag="VOX Announcement")
-
-	for(var/word in words)
-		play_vox_word(word, src.z, null)
-
-
-/proc/play_vox_word(word, z_level, mob/only_listener)
-
-	word = lowertext(word)
-
-	if(GLOB.vox_sounds[word])
-
-		var/sound_file = GLOB.vox_sounds[word]
-		var/sound/voice = sound(sound_file, wait = 1, channel = CHANNEL_VOX)
-		voice.status = SOUND_STREAM
-
-	// If there is no single listener, broadcast to everyone in the same z level
+	log_game("[key_name(src)] started making a 15.AI announcement with the following message: [message]")
+	message_admins("[key_name(src)] started making a 15.AI announcement with the following message: [message]")
+	play_vox_word(message, character_to_use, src, src.z, null)
+
+
+/proc/play_vox_word(message, character, mob/living/silicon/ai/speaker, z_level, mob/only_listener)
+	var/api_url = "https://api.15.ai/app/getAudioFile"
+	var/static/vox_voice_number = 0
+	var/datum/http_request/req = new()
+	vox_voice_number++
+	req.prepare(RUSTG_HTTP_METHOD_POST, api_url, "{\"character\":\"[character]\",\"text\":\"[message]\",\"emotion\":\"Contextual\"}", list("Content-Type" = "application/json", "User-Agent" = "/tg/station 13 server"), json_encode(list("output_filename" = "data/vox_[vox_voice_number].wav")))
+	req.begin_async()
+	UNTIL(req.is_complete())
+	var/datum/http_response/res = req.into_response()
+	if(res.status_code == 200)
+		var/full_name_file = "data/vox_[vox_voice_number].wav"
+		shell("./data/ffmpeg.exe -nostats -loglevel 0 -i ./[full_name_file] -vn -y ./data/vox_[vox_voice_number].mp3")
+		if (!istype(SSassets.transport, /datum/asset_transport/webroot))
+			log_game("CDN not set up, VOX aborted.")
+			message_admins("CDN not set up, VOX aborted.")
+			fdel("data/vox_[vox_voice_number].wav")
+			fdel("data/vox_[vox_voice_number].mp3")
+			return
+		var/datum/asset_transport/webroot/WR = SSassets.transport
+		var/shit_fuck_ass = "data/vox_[vox_voice_number].mp3"
+		var/datum/asset_cache_item/ACI = new("[md5filepath(shit_fuck_ass)].mp3", file("data/vox_[vox_voice_number].mp3"))
+		ACI.namespace = "15aivox"
+		WR.save_asset_to_webroot(ACI)
+		var/url = WR.get_asset_url(null, ACI)
+
+		log_game("[key_name(speaker)] finished making a 15.AI announcement with the following message: [message]")
+		message_admins("[key_name(speaker)] finished making a 15.AI announcement with the following message: [message]")
+		speaker.say(";[message]")
+ 		// If there is no single listener, broadcast to everyone in the same z level
 		if(!only_listener)
 			// Play voice for all mobs in the z level
 			for(var/mob/M in GLOB.player_list)
-				if(M.can_hear() && (M.client.prefs.toggles & SOUND_ANNOUNCEMENTS))
-					var/turf/T = get_turf(M)
-					if(T.z == z_level)
-						SEND_SOUND(M, voice)
+				var/turf/T = get_turf(M)
+				if(T.z == z_level && M.can_hear())
+					to_chat(M, "<audio autoplay><source src=\"[url]\" type=\"audio/mpeg\"></audio>")
```

Comment:
Caching means the deleting trick won't even actually work to solve this issue.

both the browser and cloudflare cache files, but not on testing because that would make it a pain to test changes to the asset cache system, and thats what i originally made the python file for.

Reply:
Resolved.